src/ejabberd_c2s.erl

```diff
@@ -1135,8 +1141,24 @@ handle_incoming_message({broadcast, Type, From, Packet}, StateName, StateData) -
     lists:foreach(fun(USR) -> ejabberd_router:route(From, jid:make(USR), Packet) end,
         lists:usort(Recipients)),
     fsm_next_state(StateName, StateData);
+handle_incoming_message({call_info_handler, Tag, Data}, StateName, StateData) ->
+    NStateData =
+    case ejabberd_c2s_info_handler:get_for(Tag, StateData) of
+        {Tag, HandlerState} ->
+            case ejabberd_c2s_info_handler:safe_call(Tag, Data, HandlerState, StateData) of
+                {error, _} ->
+                    StateData;
+                NStateData0 ->
+                    NStateData0
+            end;
+        no_handler ->
+            ?INFO_MSG("event=no_info_handler tag=~p host=~p", [Tag, ejabberd_c2s_state:server(StateData)]),
+            StateData
+    end,
+    fsm_next_state(StateName, NStateData);
 handle_incoming_message(Info, StateName, StateData) ->
-    ?ERROR_MSG("Unexpected info: ~p", [Info]),
+    ?ERROR_MSG("event=unexpected_info info=~p state_name=~p host=~p",
```

Comment:
I noticed we have a small inconsistency in the key=value log formatting. Here we separate `key=value` pairs with space only, and in other modules with just a `,` and no space. I don't have a strong opinion which formatting to use other than use the one which is most common in the entire codebase.

Reply:
space wins with coma, the score is 51:26.